justfile

```diff
@@ -440,7 +454,75 @@ _linkerd-viz-uninit:
 
 
 # TODO linkerd-jaeger-install
-# TODO linkerd-multicluster-install
+
+##
+## linkerd multicluster
+## 
+
+# TODO: read from multicluster values.yaml
+_pause_image := "gcr.io/google_containers/pause:3.2"
+
+_mc-target-k3d-flags := "--k3s-arg --disable='local-storage,metrics-server@server:*'"
+
+linkerd-mc-install: _mc-init _linkerd-init
+    {{ just_executable() }} \
+        k3d-name='{{ k3d-name }}-target' \
+        k3d-k8s='{{ k3d-k8s }}' \
+        k3d-servers='{{ k3d-servers }}' \
+        k3d-agents='{{ k3d-agents }}' \
+        k3d-network='{{ k3d-network }}' \
+        _k3d-flags='{{ _mc-target-k3d-flags }}' \
+        linkerd-tag='{{ linkerd-tag }}' \
+        controller-image='{{ controller-image }}' \
+        proxy-image='{{ proxy-image }}' \
+        proxy-init-image='{{ proxy-init-image }}' \
+        linkerd-exec='{{ linkerd-exec }}' \
+        _linkerd-init
+    {{ _linkerd }} mc install --set='linkerdVersion={{ linkerd-tag }}' \
+        | {{ _kubectl }} apply -f -
+    {{ linkerd-exec }} --context='k3d-{{ k3d-name }}-target' mc install \
+        --set='linkerdVersion={{ linkerd-tag }}' \
+        | kubectl --context='k3d-{{ k3d-name }}-target' apply -f -
+
+# Wait for all test namespaces to be removed before uninstalling linkerd-multicluster.
+linkerd-mc-uninstall:
+    {{ _linkerd }} mc  uninstall | {{ _kubectl }} delete -f -
+    {{ linkerd-exec }} --context='k3d-{{ k3d-name }}-target' mc uninstall \
+        kubectl --context='k3d-{{ k3d-name }}-target' delete -f - 
+
+# Ensure the default (source) cluster is initialized, and then ensure
+# that an additional (target) cluster is created with a loadbalancer in front of it.
+_mc-init: _k3d-init
+    {{ just_executable() }} \
+        k3d-k8s='{{ k3d-k8s }}' \
+        k3d-name='{{ k3d-name }}-target' \
+        k3d-network='{{ k3d-network }}' \
+        _k3d-flags='{{ _mc-target-k3d-flags }}' \
+        _k3d-init
+
+_mc-load: _mc-init linkerd-viz-load
+    if [ -z "$(docker image ls -q '{{ _pause_image }}')" ]; then docker pull -q '{{ _pause_image }}';fi
+    k3d image import --mode=direct --cluster='{{ k3d-name }}-target' {{ _pause_image }}
+    {{ just_executable() }} \
+       k3d-name='{{ k3d-name }}-target' \
+       linkerd-viz-load
+
+_linkerd-mc-ready:
+    {{ _kubectl }} wait pod --for=condition=ready \
+        --namespace=linkerd-multicluster  \
+        --timeout=1m
+    kubectl --context='k3d-{{ k3d-name }}-target' wait pod --for=condition=ready \
+        --namespace=linkerd-multicluster \
+        --timeout=1m
+
+export GO111MODULE := "on"
+
+mc-test: _mc-init _mc-load
```

Comment:
In this case, this `_mc-init` is redundant, since `_mc-load` invokes it as a dependency.